Modules/itertoolsmodule.c

```diff
@@ -4751,19 +4729,47 @@ PyInit_itertools(void)
     };
 
     Py_SET_TYPE(&teedataobject_type, &PyType_Type);
-    m = PyModule_Create(&itertoolsmodule);
-    if (m == NULL) {
-        return NULL;
-    }
 
-    for (i=0 ; typelist[i] != NULL ; i++) {
+    for (int i=0 ; typelist[i] != NULL ; i++) {
```

Comment:
While you are refactoring this code, what do you think of adding a type variable to avoid repeating `typelist[i]`?

```
PyTypeObject *type = typelist[i];
```